chainer/functions/math/matmul.py

```diff
@@ -154,3 +156,82 @@ def matmul(a, b, transa=False, transb=False):
 
     """
     return MatMul(transa=transa, transb=transb)(a, b)
+
+
+def _get_size(typ, index):
+    if index == 2 and type_check.eval(typ.ndim) == 2:
+        return 1
+    else:
+        return typ.shape[index]
+
+
+def _batch_matmul(a, b, transa, transb, transout):
+    a = a.reshape(a.shape[:2] + (-1,))
+    b = b.reshape(b.shape[:2] + (-1,))
+    return _matmul(a, b, transa, transb, transout)
+
+
+class BatchMatMul(function.Function):
+
+    def __init__(self, transa=False, transb=False):
+        self.transa = transa
+        self.transb = transb
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 2)
+        a_type, b_type = in_types
+
+        type_check.expect(
+            a_type.dtype == numpy.float32,
```

Comment:
While `MatMul`'s typecheck is `a_type.dtype.kind == 'f'`, it's also fine to restrict to float32, because `BatchMatMul` will revive for compatibility.  Do you have ideas?

Reply:
Let's expand acceptable dtype

Reply:
This function is deprecated. Thus, I think that change is unnecessary